pkg/search/postgres/query/uuid_query.go

```diff
@@ -0,0 +1,108 @@
+	if err != nil {
+		return nil, err
+	}
+	return qeWithSelectFieldIfNeeded(ctx, &whereClause, nil), nil
+}
+
+func newUUIDQueryWhereClause(columnName string, value string, queryModifiers ...pkgSearch.QueryModifier) (WhereClause, error) {
+	if len(value) == 0 {
+		return WhereClause{}, errors.New("value in search query cannot be empty")
+	}
+
+	if value == pkgSearch.WildcardString {
+		return WhereClause{
+			Query:  fmt.Sprintf("%s is not null", columnName),
+			Values: []interface{}{},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				foundVal := strings.ToLower(foundValue.(string))
+				return foundVal != ""
+			},
+		}, nil
+	}
+
+	if len(queryModifiers) == 0 {
+		uuidVal, err := uuid.FromString(value)
+		if err != nil {
+			return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+		}
+		return WhereClause{
+			Query:  fmt.Sprintf("%s = $$", columnName),
+			Values: []interface{}{uuidVal},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				return strings.EqualFold(foundValue.(string), value)
+			},
+		}, nil
+	}
+	if queryModifiers[0] == pkgSearch.AtLeastOne {
+		panic("I dont think this is used")
+	}
+	var negationString string
+	negated := queryModifiers[0] == pkgSearch.Negation
+	if negated {
+		negationString = "!"
+		if len(queryModifiers) == 1 {
+			uuidVal, err := uuid.FromString(value)
+			if err != nil {
+				return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+			}
+			return WhereClause{
+				Query:  fmt.Sprintf("%s != $$", columnName),
+				Values: []interface{}{uuidVal},
+				equivalentGoFunc: func(foundValue interface{}) bool {
+					return strings.EqualFold(foundValue.(string), value) != negated
+				},
+			}, nil
+		}
+		queryModifiers = queryModifiers[1:]
+	}
+
+	switch queryModifiers[0] {
+	case pkgSearch.Regex:
+		re, err := regexp.Compile(value)
+		if err != nil {
+			return WhereClause{}, fmt.Errorf("invalid regexp %s: %w", value, err)
+		}
+
+		if value != "*" {
+			return WhereClause{}, fmt.Errorf("invalid regexp %s for UUID field", value)
+		}
+		return WhereClause{
+			Query:  fmt.Sprintf("%s is not null", columnName),
+			Values: []interface{}{value},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				foundVal := strings.ToLower(foundValue.(string))
+				return re.MatchString(foundVal) != negated
+			},
+		}, nil
+	case pkgSearch.Equality:
+		uuidVal, err := uuid.FromString(value)
+		if err != nil {
+			return WhereClause{}, errors.Wrapf(err, "value %q in search query must be valid UUID", value)
+		}
+		return WhereClause{
+			Query:  fmt.Sprintf("%s %s= $$", columnName, negationString),
+			Values: []interface{}{uuidVal},
+			equivalentGoFunc: func(foundValue interface{}) bool {
+				return strings.EqualFold(foundValue.(string), value) != negated
+			},
+		}, nil
+	}
+	err := fmt.Errorf("unknown query modifier: %s", queryModifiers[0])
+	utils.Should(err)
```

Comment:
Since we already use the switch case, let's use this within a default case.
```suggestion
        default:
	        err := fmt.Errorf("unknown query modifier: %s", queryModifiers[0])
	        utils.Should(err)
	        return WhereClause{}, err
```

Reply:
I recently learned that we should use `utils.Should` only in case or programatic errors that are not tied to user input. In this case, I'm not sure we should use it (since its bound on the query modifier parameter from the consumer of the function) - so it might be just sufficient enough to log within the default case and return an error here instead of using `utils.Should`.